browser-test/src/civiform_admin_question_list.test.ts

```diff
@@ -0,0 +1,123 @@
+    // of the questions added in this test.
+    // TODO(#3029): Consider asserting on the whole list if it can be considered stable even in
+    // prober environments and with canonical question creation.
+
+    // A question cannot be published in isolation. In order to allow making created questions
+    // active, create a fake program.
+    const programName = 'question list test program'
+    await adminPrograms.addProgram(programName)
+
+    // Most recently added question is on top.
+    await expectQuestionListTopElements(adminQuestions, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+
+    // Publish all programs and questions, order should be maintained.
+    await adminPrograms.publishProgram(programName)
+    // Create a draft version of the program so that the question bank can be accessed.
+    await adminPrograms.createNewVersion(programName)
+    await expectQuestionListTopElements(adminQuestions, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionTwoPublishedText,
+      questionOnePublishedText,
+    ])
+
+    // Now create a draft version of the previously last question. After,
+    // it should be on top.
+    await adminQuestions.createNewVersion(questionOnePublishedText)
+    const questionOneDraftText = `${questionOnePublishedText} new version`
+    await expectQuestionListTopElements(adminQuestions, [
+      questionOneDraftText,
+      questionTwoPublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionOnePublishedText,
+      questionTwoPublishedText,
+    ])
+
+    // Now create a new question, which should be on top.
+    const questionThreePublishedText = 'question list test question three'
+    await adminQuestions.addNameQuestion({
+      questionName: questionThreePublishedText,
+      questionText: questionThreePublishedText,
+    })
+    await expectQuestionListTopElements(adminQuestions, [
+      questionThreePublishedText,
+      questionOneDraftText,
+      questionTwoPublishedText,
+    ])
+    await expectQuestionBankTopElements(programName, adminPrograms, [
+      questionThreePublishedText,
+      questionOnePublishedText,
+      questionTwoPublishedText,
+    ])
+  })
+
+  async function expectQuestionListTopElements(
+    adminQuestions: AdminQuestions,
+    expectedQuestions: string[],
+  ) {
+    if (!expectedQuestions) {
+      throw new Error('expected at least one question')
+    }
+    const questionListNames = await adminQuestions.questionNames()
+    expect(questionListNames.length).toBeGreaterThanOrEqual(
+      expectedQuestions.length,
+    )
+    expect(questionListNames.slice(0, expectedQuestions.length)).toEqual(
+      expectedQuestions,
+    )
```

Comment:
I think that's the question from the previous review, don't remember the answer. Why can't we assume that number of questions strictly matches expectedQuestions given that test wipes out DB at the beginning of each test file.

Reply:
I believe we likely can, but wanted to separate that from this change (since the pre-existing `admin_program_list.test.ts` follows the same pattern.

I left a TODO on line 29 to track cleaning this up in a follow-on PR if possible. At worst, I'm creating extra work for myself. Does that sound like an acceptable way forward to you?